torch_frame/data/tensor_frame.py

```diff
@@ -168,26 +169,28 @@ def __eq__(self, other: Any) -> bool:
         if self.col_names_dict != other.col_names_dict:
             return False
         # Match feat_dict
-        for stype_name in self.feat_dict.keys():
-            self_feat = self.feat_dict[stype_name]
+        for stype_name, self_feat in self.feat_dict.items():
             other_feat = other.feat_dict[stype_name]
-            if type(self_feat) != type(other_feat):
-                return False
-            if not isinstance(self_feat, dict):
+            if isinstance(self_feat, Tensor):
+                if not isinstance(other_feat, Tensor):
+                    return False
                 if self_feat.shape != other_feat.shape:
                     return False
                 if not torch.allclose(self_feat, other_feat):
                     return False
-            else:
-                for key in self_feat:
-                    self_feat_value = self_feat[key]
-                    other_feat_value = other_feat[key]
-                    if self_feat_value.shape != other_feat_value.shape:
-                        return False
-                    # TODO: Support allclose for MultiNestedTensor
-                    if isinstance(self_feat_value,
-                                  Tensor) and not torch.allclose(
-                                      self_feat_value, other_feat_value):
+            elif isinstance(self_feat, MultiNestedTensor):
+                if not isinstance(other_feat, MultiNestedTensor):
+                    return False
+                if not MultiNestedTensor.allclose(self_feat, other_feat):
+                    return False
+            elif isinstance(self_feat, Dict):
+                if not isinstance(other_feat, Dict):
```

Comment:
QQ: Any idea what the difference from built-in `dict` is when evaluating object with `isinstance`?
```suggestion
            elif isinstance(self_feat, dict):
                if not isinstance(other_feat, dict):
```

Reply:
I think they have the same behavior.
```python
>>> from typing import Dict
>>> isinstance({}, Dict)
True
>>> isinstance({}, dict)
True
>>>
```

Reply:
(cc @yiweny as I remember you also left a similar comment on my previous PR: `isinstance(obj, Tuple)` vs. `isinstance(obj, tuple)`.)